### ISSUE TITLE
[Bug]: [Nightly]Large numbers of failed cases occurs with error message: find no available datacoord

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#18806]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master(9338ad4)
- Deployment mode(standalone or cluster):cluster(pulsar)
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.2.0.dev19
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Large numbers of failed cases occurs with error message: find no available datacoord .
Start at about 2022-08-23T15:29:40.530Z

```
[2022-08-23T15:29:40.530Z] [gw3] [ 44%] FAILED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_guarantee_timestamp[\u4e2d\u6587] 
[2022-08-23T15:29:47.056Z] [gw1] [ 44%] XFAIL testcases/test_search.py::TestCollectionSearch::test_search_before_after_delete[500-128-True-False] 
[2022-08-23T15:29:47.057Z] [gw0] [ 44%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_after_index_different_metric_type[8-False-False-HNSW-params4] 
[2022-08-23T15:30:01.888Z] testcases/test_search.py::TestCollectionSearch::test_search_after_index_different_metric_type[8-False-False-ANNOY-params5] 
[2022-08-23T15:30:01.890Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_guarantee_timestamp[aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa] 
[2022-08-23T15:30:01.890Z] testcases/test_search.py::TestCollectionSearch::test_search_before_after_delete[500-128-True-True] 
[2022-08-23T15:30:01.890Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[] 
[2022-08-23T15:30:19.923Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[] 
[2022-08-23T15:30:19.924Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[ ] 
[2022-08-23T15:30:23.185Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[ ] 
[2022-08-23T15:30:23.186Z] [gw1] [ 44%] XFAIL testcases/test_search.py::TestCollectionSearch::test_search_before_after_delete[500-128-True-True] 
[2022-08-23T15:30:41.223Z] testcases/test_search.py::TestCollectionSearch::test_search_collection_after_release_load[3000-2-8-False-False] 
[2022-08-23T15:30:41.224Z] [gw5] [ 44%] FAILED testcases/test_query.py::TestQueryParams::test_query_expr_none 
[2022-08-23T15:31:03.109Z] [gw4] [ 44%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_after_different_index_with_params[8-False-True-RHNSW_PQ-params7] 
[2022-08-23T15:31:03.110Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[12-s] 
[2022-08-23T15:31:21.147Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[12-s] 
[2022-08-23T15:31:21.148Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[12 s] 
[2022-08-23T15:31:21.148Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[12 s] 
[2022-08-23T15:31:21.148Z] testcases/test_query.py::TestQueryParams::test_query_non_string_expr 
[2022-08-23T15:31:43.018Z] [gw3] [ 44%] FAILED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_guarantee_timestamp[aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa] 
[2022-08-23T15:31:43.020Z] testcases/test_search.py::TestCollectionSearch::test_search_after_different_index_with_params[8-False-True-RHNSW_SQ-params8] 
[2022-08-23T15:31:43.020Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[(mn)] 
[2022-08-23T15:31:43.020Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[(mn)] 
[2022-08-23T15:32:01.050Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_invalid_round_decimal[7] 
[2022-08-23T15:32:01.052Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[\u4e2d\u6587] 
[2022-08-23T15:32:01.052Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[\u4e2d\u6587] 
[2022-08-23T15:32:22.921Z] [gw0] [ 44%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_after_index_different_metric_type[8-False-False-ANNOY-params5] 
[2022-08-23T15:32:22.923Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[%$#] 
[2022-08-23T15:32:22.923Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[%$#] 
[2022-08-23T15:32:22.923Z] [gw5] [ 44%] FAILED testcases/test_query.py::TestQueryParams::test_query_non_string_expr 
[2022-08-23T15:32:22.923Z] testcases/test_search.py::TestCollectionSearch::test_search_after_index_different_metric_type[8-False-False-RHNSW_FLAT-params6] 
[2022-08-23T15:32:22.923Z] testcases/test_query.py::TestQueryParams::test_query_expr_invalid_string 
[2022-08-23T15:32:22.923Z] [gw1] [ 44%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_collection_after_release_load[3000-2-8-False-False] 
[2022-08-23T15:32:40.952Z] testcases/test_search.py::TestCollectionSearch::test_search_collection_after_release_load[3000-2-8-False-True] 
[2022-08-23T15:32:40.954Z] testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa] 
[2022-08-23T15:32:41.209Z] [gw2] [ 44%] SKIPPED testcases/test_search.py::TestCollectionSearchInvalid::test_search_param_invalid_field_type[aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa] 
[2022-08-23T15:33:03.186Z] [gw4] [ 44%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_after_different_index_with_params[8-False-True-RHNSW_SQ-params8] 
```

### Expected Behavior

PASSED

### Steps To Reproduce

```markdown
See nightly log:
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/54/pipeline/171
```

### Milvus Log

1. milvus log:https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/54/artifacts
or download the log In the comments below.

2. error:
[ci.var.log.containers.mdp-54-n-milvus-indexcoord-6cfd9458cf-6s8kw_milvus-ci_indexcoord-82a623e935084b3562be78eef457e2197fb56b50ef4562410bc6dd3d3e680e57.log]
[ci.var.log.containers.mdp-54-n-milvus-querycoord-84f95b9b-4cqh8_milvus-ci_querycoord-258172a517b0886478e68bf5c8c48c0c22c45cf6382fc0978c87765581eec287.log]
[ci.var.log.containers.mdp-54-n-milvus-querycoord-84f95b9b-4cqh8_milvus-ci_querycoord-f567f15d6c64ca5aeb9b1548068068aa59fae5cdcafe9493356c88291e5aabe7.log]

```
2022-08-23T23:01:52.908753534+08:00 stderr F panic: All attempts results:
2022-08-23T23:01:52.908767407+08:00 stderr F attempt #1:err: find no available datacoord, check datacoord state
2022-08-23T23:01:52.908772257+08:00 stderr F , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
2022-08-23T23:01:52.908777191+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:259 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall
2022-08-23T23:01:52.90878445+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:122 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetComponentStates
2022-08-23T23:01:52.908789162+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:67 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1
2022-08-23T23:01:52.90879341+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 github.com/milvus-io/milvus/internal/util/retry.Do
2022-08-23T23:01:52.908800105+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:91 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates
2022-08-23T23:01:52.908804525+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:106 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy
2022-08-23T23:01:52.908809242+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:139 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).init
2022-08-23T23:01:52.908814124+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:79 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Run
2022-08-23T23:01:52.908818443+08:00 stderr F /go/src/github.com/milvus-io/milvus/cmd/components/index_coord.go:48 github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Run
2022-08-23T23:01:52.908822702+08:00 stderr F
2022-08-23T23:01:52.908826842+08:00 stderr F attempt #2:err: find no available datacoord, check datacoord state
2022-08-23T23:01:52.908831932+08:00 stderr F , /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
2022-08-23T23:01:52.908835772+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:259 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall
2022-08-23T23:01:52.908839994+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:122 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).GetComponentStates
2022-08-23T23:01:52.908846901+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:67 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1
2022-08-23T23:01:52.908853207+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 github.com/milvus-io/milvus/internal/util/retry.Do
2022-08-23T23:01:52.908857119+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:91 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates
2022-08-23T23:01:52.908864978+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:106 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy
2022-08-23T23:01:52.908869409+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:139 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).init
2022-08-23T23:01:52.908873907+08:00 stderr F /go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:79 github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Run
2022-08-23T23:01:52.908878652+08:00 stderr F /go/src/github.com/milvus-io/milvus/cmd/components/index_coord.go:48 github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Run
2022-08-23T23:01:52.908882392+08:00 stderr F
2022-08-23T23:01:52.908886861+08:00 stderr F attempt #3:err: find no available datacoord, check datacoord state
```

**Comment by NicoYuan1986** [comment]
The file is too large to upload. Contact me for milvus log if the website is unreachable.

**Comment by soothing-rain** [comment]
The error as instructed in the title is not the root cause.

The root cause could be errors like:
```
[querynode/segment_loader.go:88] ["find no valid segment target, skip load segment"] [request="base:<msg_type:LoadSegments msgID:435491626074651192 > schema:<name:\"search_collection_1wteWmt2\" fields:<fieldID:100 name:\"int64\" is_primary_key:true data_type:Int64 > fields:<fieldID:101 name:\"float\" data_type:Float > fields:<fieldID:102 name:\"varchar\" data_type:VarChar type_params:<key:\"max_length\" value:\"65535\" > > fields:<fieldID:103 name:\"float_vector\" data_type:FloatVector type_params:<key:\"dim\" value:\"128\" > > > collectionID:435491651489366017 load_meta:<load_type:LoadCollection collectionID:435491651489366017 partitionIDs:435491651489366018 > "]
```

/assign @congqixia 
Do you think this would be resolved in 2.2?

**Comment by NicoYuan1986** [comment]
This issue no longer occurs in master(c58a8d8).
